Tests/SentryTests/Networking/SentryHTTPInterceptorTests.swift

```diff
@@ -0,0 +1,256 @@
+        
+        let spanHeader = fixture.span.toTraceHeader()
+        XCTAssertNotEqual(request, newRequest)
+        XCTAssertEqual(spanHeader.value(), newRequest.value(forHTTPHeaderField: SENTRY_TRACE_HEADER))
+        
+        let interceptFlag = URLProtocol.property(forKey: SENTRY_INTERCEPTED_REQUEST, in: newRequest) as? NSNumber
+        
+        XCTAssertTrue(interceptFlag!.boolValue)
+    }
+    
+    func testCanonicalRequestForRequestWithoutTransaction() {
+        fixture.scope.span = nil
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        let newRequest = SentryHttpInterceptor.canonicalRequest(for: request)
+        
+        XCTAssertEqual(request, newRequest)
+        XCTAssertNil(newRequest.value(forHTTPHeaderField: SENTRY_TRACE_HEADER))
+    }
+    
+    func testIgnoreAlreadyInterceptedRequest() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        let newRequest = SentryHttpInterceptor.canonicalRequest(for: request)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: newRequest))
+    }
+    
+    func testProtocolGetRegisteredAndUnregistered() {
+        NSURLProtocolSwizzle.swizzleURLProtocol()
+        NSURLProtocolSwizzle.shared.registerCallback = { protocolClass in
+            XCTAssertTrue(protocolClass === SentryHttpInterceptor.self)
+        }
+        NSURLProtocolSwizzle.shared.unregisterCallback = { protocolClass in
+            XCTAssertTrue(protocolClass === SentryHttpInterceptor.self)
+        }
+        let integration = SentryNetworkTrackingIntegration()
+        integration.install(with: fixture.options)
+        integration.uninstall()
+        
+        NSURLProtocolSwizzle.shared.registerCallback = nil
+        NSURLProtocolSwizzle.shared.unregisterCallback = nil
+    }
+    
+    func testProtocolDontGetRegistered() {
+        NSURLProtocolSwizzle.swizzleURLProtocol()
+        NSURLProtocolSwizzle.shared.registerCallback = { _ in
+            XCTAssert(false)
+        }
+        let integration = SentryNetworkTrackingIntegration()
+        fixture.options.enableAutoPerformanceTracking = false
+        integration.install(with: fixture.options)
+        integration.uninstall()
+    }
+    
+    func testCreateSession() {
+        let interceptor = SentryHttpInterceptor(request: fixture.request, cachedResponse: nil, client: nil)
+        let session = interceptor.createSession()
+        XCTAssertNotNil(session)
+        XCTAssertTrue(session.delegate === interceptor)
+        XCTAssertEqual(session.configuration, URLSessionConfiguration.default)
+    }
+    
+    func testStartLoading() {
+        let now = Date()
+        fixture.dateProvider.setDate(date: now)
+        
+        let sut = fixture.getSut()
+        sut.startLoading()
+        let session = sut.session as? TestURLSession
+        
+        XCTAssertEqual(session?.lastDataTask?.resumeDate, now)
```

Comment:
`l`: Why not like this? Same applies for `testStopLoading`.
```suggestion
        let sut = fixture.getSut()
        sut.startLoading()
        let session = sut.session as? TestURLSession
        
        XCTAssertEqual(session?.lastDataTask?.resumeDate, fixture.dateProvider.date())
```

Reply:
After the SentrySessionTest error I'm just worried about the date being changed elsewhere.